server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMessagesMethod.java

```diff
@@ -90,73 +93,83 @@
         Preconditions.checkArgument(request instanceof GetMessagesRequest);
         GetMessagesRequest getMessagesRequest = (GetMessagesRequest) request;
         Optional<ImmutableSet<MessageProperty>> requestedProperties = getMessagesRequest.getProperties();
+        Optional<ImmutableSet<MessageHeaderProperty>> headerProperties = getMessagesRequest.getHeaderProperties();
```

Comment:
Request objects is supposed to match the specification and I guess HeaderProperties is not part of it.